src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,206 @@
+package duke;
+
+import duke.task.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@code TaskList} class encapsulates a list of tasks
+ * to keep track of
+ */
+public class TaskList {
+
+    /**
+     * a list that keeps track of current and new tasks
+     */
+    private ArrayList<Task> storage;
+
+
+    /**
+     * Constructor method for TaskList class
+     * @param lines list of lines that convert into tasks, if not empty
+     */
+    public TaskList(List<String> lines) {
+        this.storage = new ArrayList<>(100);
+        if (lines.size() != 0){
+            fillStorage(lines);
+        }
+    }
+
+    /**
+     * Converts each line of {@code String} to {@code Task} object
+     * before being added into taskList
+     * @param lines list of lines read from file
+     */
+    private void fillStorage(List<String> lines) {
+        for (String line: lines) {
+            String[] arguments = line.split (" | ");
+            switch(arguments[1]) {
+                case "T":
+                    storeTodoTask(arguments);
+                    break;
+                case "D":
+                    storeDeadlineTask(arguments);
+                    break;
+                case "E":
+                    storeEventTask(arguments);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * stores ToDo tasks into taskList NOSONAR
+     * @param args array of arguments from a line
+     */
+    private void storeTodoTask(String[] args) {
+        Task taskToAdd = new ToDo(args[3].trim());
+        if (args[2].contains("X")) {
+            taskToAdd.markAsDone();
+        }
+
+        storage.add(taskToAdd);
+    }
+
+    /**
+     * stores Deadline tasks into taskList
+     * @param args array of arguments from a line
+     */
+    private void storeDeadlineTask(String[] args){
+        String desc =  args[3].substring(0, args[3].indexOf("(by:")).trim();
+        String deadline = args[3].substring(args[3].indexOf("(by:")
+                    + "(by:".length(), args[3].indexOf(")")).trim();
+
+        Task taskToAdd = new Deadline(desc, deadline);
+        if (args[2].contains("X")){
+            taskToAdd.markAsDone();
+        }
+
+        storage.add(taskToAdd);
+    }
+
+    /**
+     * stores Event tasks into taskList
```

Comment:
Capitalization should be used for consistency but otherwise, your docs written are clear and concise!